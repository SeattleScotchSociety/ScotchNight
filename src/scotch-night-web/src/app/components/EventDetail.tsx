import { inject, observer } from "mobx-react";
import * as React from "react";

import { IRootStore } from "../stores/RootStore";
import { AddBottle } from "./AddBottle";
import { GenericMap, IMarker } from "./GenericMap";
import { Modal } from "./Modal";
import { TastingMenu } from "./TastingMenu";

import * as format from "date-fns/format";

interface IEventDetailProps { store: IRootStore; }
interface IEventDetailState {
    position: {
        lat: string,
        lng: string
    };
}

@inject("store")
@observer
export class EventDetail extends React.Component<IEventDetailProps, IEventDetailState> {
    private add: AddBottle;

    constructor(props: IEventDetailProps) {
        super(props);

        this.buildEventMap = this.buildEventMap.bind(this);
        this.handleOnAddBottle = this.handleOnAddBottle.bind(this);
    }

    private handleOnAddBottle() {
        const { scotchNightStore } = this.props.store;
        const { navigation } = this.props.store;
        navigation.push(`${scotchNightStore.currentEvent.id}/add-bottle`);
    }

    private buildEventMap(position: any) {
        let center = {
            lat: 47.6062095,
            lng: -122.3320708
        };

        if (position) {
            center = { ...position };
        }

        const markers = [{
            key: "event-marker",
            position: center
        }];

        return (
            <GenericMap
                containerElement={
                    <div style={{ height: `100%` }} />
                }
                mapElement={
                    <div style={{ height: `100%` }} />
                }
                ref={() => null}
                defaultZoom={13}
                defaultCenter={center}
                onClick={() => null}
                markers={markers}
            />
        );
    }

    public render() {
        const { scotchNightStore } = this.props.store;
        const { currentEvent } = scotchNightStore;

        if (!currentEvent) {
            return null;
        }

        const { date, description, location, title } = currentEvent;
        const { position } = location;
        const eventMap = this.buildEventMap(position);

        return (
            <div className="container">
                <h1>{title}</h1>
                <p className="margin-vertical-md"><em>{description}</em></p>
                <p className="event__detail"><i className="fa fa-fw fa-calendar-o" />&nbsp;{format(date, "MMMM D, YYYY")}</p>
                <p className="event__detail"><i className="fa fa-fw fa-clock-o" />&nbsp;{format(date, "h:mm aa")}</p>
                <p className="event__detail"><i className="fa fa-fw fa-map-marker" />&nbsp;{location.name}</p>
                {eventMap}
                <TastingMenu store={this.props.store} />
                <button className="btn btn--primary btn--block" onClick={this.handleOnAddBottle}>Add Bottle</button>
                <AddBottle ref={(add) => { this.add = add; }} store={this.props.store} />
            </div>
        );
    }
}

export default EventDetail;
