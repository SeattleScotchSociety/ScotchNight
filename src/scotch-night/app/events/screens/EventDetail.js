// @flow
import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { List, ListItem, Button } from 'react-native-elements';
import { connect } from 'react-redux';
import { SimpleLineIcons } from '@expo/vector-icons';
import * as moment from 'moment';
import _ from 'lodash';

function TastingMenu(props) {
    let { menu, onPress } = props;

    if (!menu || menu.length === 0) {
        return (<Text style={styles.noBottles}>No Bottles</Text>);
    }

    let menuList = menu.map((item, index) => {
        return (
            <ListItem key={index} title={`${item.distillery} ${item.name}`} onPress={onPress.bind(null, item)} />
        );
    });

    return (
        <List>{menuList}</List>
    );
}

class EventDetail extends Component {
    constructor(props) {
        super(props);

        this._handleOnPress = this._handleOnPress.bind(this);
        this._handleOnAddBottle = this._handleOnAddBottle.bind(this);
    }

    _handleOnPress(bottle) {
        let { navigate } = this.props.navigation;
        navigate('Bottle', { bottle: bottle });
    }

    _handleOnAddBottle() {
        let { navigate } = this.props.navigation;
        navigate('AddBottle', { event: this.props.navigation.state.params });
    }

    render() {
        let { menu } = this.props;
        let { title, location, description, date } = this.props.event;

        return (
            <View style={styles.container}>
                <View style={styles.eventContainer}>
                    <Text style={styles.eventTitle}>{title}</Text>
                    <View style={{ flexDirection: 'row', marginLeft: 15, marginBottom: 5 }}>
                        <View style={{ flexDirection: 'row', marginRight: 10 }}>
                            <SimpleLineIcons name='location-pin' color='#80807f' size={16} />
                            <Text style={styles.eventDetail}>{location}</Text>
                        </View>
                        <View style={{ flexDirection: 'row' }}>
                            <SimpleLineIcons name='calendar' color='#80807f' size={16} />
                            <Text style={styles.eventDetail}>{moment.default(date).format('dddd, MMMM, YYYY, h:mm')}</Text>
                        </View>
                    </View>
                    <Text style={styles.description}>{description}</Text>
                </View>
                <Text style={styles.menuHeader}>Tasting Menu</Text>
                <TastingMenu menu={menu} onPress={this._handleOnPress} />
                <Button title='Add Bottle' backgroundColor='#00817d' style={{ marginTop: 20 }} onPress={this._handleOnAddBottle} />
            </View>
        );
    }
};

const PROVIDER_IMAGE_SIZE = 40;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff'
    },
    eventContainer: {
        flexDirection: 'column'
    },
    eventTitle: {
        marginTop: 20,
        marginLeft: 15,
        marginBottom: 5,
        color: '#00817d',
        fontWeight: 'bold',
        fontSize: 18
    },
    description: {
        marginBottom: 20,
        marginLeft: 15
    },
    eventDetail: {
        color: '#80807f',
        marginLeft: 5
    },
    menuHeader: {
        color: '#00817d',
        fontSize: 16,
        fontWeight: 'bold',
        marginTop: 10,
        marginBottom: -10,
        marginLeft: 15
    },
    noBottles: {
        marginTop: 20,
        marginLeft: 15,
        marginBottom: 20,
        color: '#80807f'
    }
});

function mapStateToProps(state) {
    let currentEvent = state.events.selected;
    let menu = [];

    if (currentEvent.bottles) {
        menu = _.filter(state.bottles, bottle => currentEvent.bottles.includes(bottle.id));
    }

    return { event: currentEvent, menu };
}

export default connect(mapStateToProps)(EventDetail);
