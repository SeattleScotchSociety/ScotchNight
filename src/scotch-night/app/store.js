import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { AsyncStorage } from 'react-native';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer } from './Reducers';
import rootSaga from './Sagas';
import Bottles from './bottles';
let { loadBottles } = Bottles.Actions;

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
    reducer,
    undefined,
    compose(applyMiddleware(sagaMiddleware), autoRehydrate())
);

sagaMiddleware.run(rootSaga);

persistStore(store, { storage: AsyncStorage }, () => {
    store.dispatch(loadBottles());
}).purge();

export default store;
