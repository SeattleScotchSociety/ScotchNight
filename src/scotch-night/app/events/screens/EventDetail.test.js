import React from 'react';
import renderer from 'react-test-renderer';
import Store from '../../store';
import EventDetail from './EventDetail';

test('renders correctly', () => {
    const tree = renderer
        .create(
            <EventDetail
                store={Store}
                navigation={{
                    state: {
                        params: {
                            event: {
                                date: '1/12/1111',
                                location: 'A Pineapple Under the Sea',
                                description: 'SPONGEBOB SQUAREPANTS!'
                            }
                        }
                    }
                }}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
