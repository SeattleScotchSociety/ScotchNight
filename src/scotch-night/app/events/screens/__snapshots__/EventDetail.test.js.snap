// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 30,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "500",
          }
        }
      >
        1/12/1111
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "marginBottom": 12,
          }
        }
      >
        A Pineapple Under the Sea
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        SPONGEBOB SQUAREPANTS!
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 30,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://www.westlanddistillery.com/assets/media/whiskeys/5e4a5c3c6cdbfabdcf67941c83f9e7d9.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 40,
          "marginRight": 10,
          "width": 40,
        }
      }
    />
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "purple",
            "fontSize": 18,
            "fontWeight": "500",
          }
        }
      >
        Adam Phillabaum
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 13,
          }
        }
      >
        June 12, 2017
      </Text>
    </View>
  </View>
</View>
`;
